dockerhub-script/action.yml

```diff
@@ -0,0 +1,40 @@
+name: Run DockerHub Script
+
+description: |
+  This composite action generates a DockerHub token and runs a script with it.
+
+inputs:
+  GPUCIBOT_DOCKERHUB_USER:
+    description: 'DockerHub username'
+    required: true
+  GPUCIBOT_DOCKERHUB_TOKEN:
+    description: 'DockerHub password'
+    required: true
+  script:
+    description: 'Script to run'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Generate DockerHub Token
+      shell: bash
+      id: generate_token
+      run: |
+        logout() {
+          curl -X POST \
+            -H "Authorization: JWT $HUB_TOKEN" \
+            "https://hub.docker.com/v2/logout/"
+        }
+
+        trap logout EXIT
+
+        HUB_TOKEN=$(
+          curl -s -H "Content-Type: application/json" \
+            -X POST \
+            -d "{\"username\": \"${GPUCIBOT_DOCKERHUB_USER}\", \"password\": \"${GPUCIBOT_DOCKERHUB_TOKEN}\"}" \
```

Comment:
Are the `GPUCIBOT_DOCKERHUB_*` inputs automatically provided as environment variables?

If not, we'll need to explicitly add them doing something like this:

```yaml
env:
  GPUCIBOT_DOCKERHUB_USER: ${{ inputs.GPUCIBOT_DOCKERHUB_USER }}
```

Reply:
Yeah, fixed.